scripts/gui/microbe_hud.mjs

```diff
@@ -384,33 +428,87 @@ function onExitToMenuClicked() {
+    }
 
-    if(vars.noCompounds){
+    const mainContent = document.createElement("div");
+    mainContent.style.width = "100%";
+    mainContent.style.height = "15px";
+    mainContent.style.color = "white";
+    mainContent.style.textAlign = "center";
+    mainContent.innerHTML = "Hello";
 
-        panel.appendChild(document.createElement("br"));
-        panel.appendChild(document.createTextNode("Nothing to eat here"));
+    if(vars.noCompounds){
 
+        mainContent.innerHTML = "Nothing to eat here";
+        panel.appendChild(mainContent);
     } else {
 
-        common.getKeys(vars).forEach(function(key){
+        if(!vars.compounds){
+            mainContent.innerHTML = "Error reading compounds Data";
+            return;
+        }
+        let objCompoundsData = null;
+
+
 
-            // Skip things that are handled elsewhere
-            if(key == "mousePos" || key == "hoveredCells")
+        if(typeof vars.compounds === "string" || vars.compounds instanceof String){
+            try{
+                objCompoundsData = JSON.parse(vars.compounds);
+            } catch(err){
+                mainContent.innerHTML = "invalid json for mouseHover info: " + err;
                 return;
+            }
+        } else {
+            objCompoundsData = vars.compounds;
+        }
+
+        // Compounds data are store in json format, so we need parse it
+        objCompoundsData = JSON.parse(vars.compounds);
+
+        mainContent.innerHTML = "At cursor:";
+        panel.appendChild(mainContent);
+
+        const title = document.createElement("p");
+        title.style.fontSize = "12pt";
+        title.style.marginTop = "0";
+        title.style.marginBottom = "5px";
+        const titleText = document.createTextNode("Compounds: ");
+        title.appendChild(titleText);
+        panel.appendChild(title);
+
 
+        // Create for each compound the information in GUI
+        objCompoundsData.forEach(function(compoundData){
             // Line breaks between elements
             panel.appendChild(document.createElement("br"));
-
-            // Debug print version
-            // Panel.appendChild(document.createTextNode(key + ": " + vars[key]));
-            panel.appendChild(document.createTextNode(vars[key]));
+            const img = document.createElement("IMG");
+            let src = "../../Textures/gui/bevel/";
+
+            src = src + unescape( compoundData.name).replace(" ", "") + ".png";
+
+            const par = document.createElement("p");
+
+            par.style.marginBottom = "0";
+            par.style.paddingBottom = "10px";
+            par.style.marginTop = "0";
+            img.setAttribute("src", src);
+            img.style.verticalAlign = "text-bottom";
+            img.setAttribute("width", "25");
+            img.setAttribute("height", "25");
+            par.appendChild(img);
+            const parText = document.createTextNode("" + compoundData.quantity);
```

Comment:
It might be nice to do the rounding here in the GUI (using toFixed).

Reply:
Unscape remove %20 from string

Reply:
Where does that come from? JSON serialize + unserialize should not affect any strings.
I'd like to fix that the string magically gets escaped somehow.

Reply:
The displayed name of compound was with %20, I try remove it with replace but didn't work, I found that method working

Reply:
If you saw that in the image src, then it makes perfect sense that the browser escaped the space. It's likely that my custom file loader in Leviathan doesn't unescape it so it can't find the actual file.
I would guess that if you remove the space (without doing any unescape call) it would work just as well and be less suspicious.

Reply:
Rounding should still be added here.